src/utils/model_utils.py

```diff
@@ -373,6 +373,40 @@ def deep_mutual_train(
             train_accuracy[i] = 100.0 * correct[i] / len(dloader.dataset)
         return train_loss, train_accuracy
 
+
+    def train_with_dpsgd(
```

Comment:
Is it any different from the original train method? If not then it would be much more ideal to avoid code duplication